### PR TITLE
Error handling

### PR DESCRIPTION
**Pull request by ashtonmoomoo**

### CODE REVIEW

File: app/app.py [hunk 1]
```diff
@@ -24,7 +25,7 @@ def question_router(question_type, question_id):
 
 @app.post(API_VERSION + "/question/<question_type>/<question_id>")
 def get_solution(question_type, question_id):
-    attempt = request.get_json()
+    attempt = request.get_json(force=True)
```
Comment (odrusso) [inline app/app.py:28]: Added this back in?

File: app/app.py [hunk 2]
```diff
@@ -35,11 +36,10 @@ def questions():
...
     return question_dict, status
 
-@app.errorhandler(QuestionNotFound)
-def question_not_found(error):
-
-    return error.json, 404
+@app.errorhandler(HttpError)
+def handle_http_error(error):
 
+    return error.json, error.json.get("status", 500)
```
Comment (odrusso) [inline app/app.py:42]: Should the status be part of the object rather than JSON?

File: app/config.py [hunk 3]
```diff
@@ -4,4 +4,4 @@
 API_VERSION = "/api/v1"
 
 DEBUG = os.environ.get("FLASK_DEBUG", "False") == "True"
-CORS_ALLOWED_ORIGINS = "https://localhost:3000" if DEBUG else ""
+CORS_ALLOWED_ORIGINS = "http://localhost:8080" if DEBUG else ""
```
Comment (odrusso) [inline app/config.py:7]: nice

File: app/app.py (hunk 1 above)
Comment (ashtonmoomoo) [reply app/app.py:28]: yeah -_- I'll need to do more research on this, just put this here to continue getting everything else working

File: app/solutions.py [hunk 4]
```diff
@@ -44,7 +32,7 @@ def check_solution(question_type, question_id, attempt):
         status = 200
 
     else:
-        attempt_response = AttemptError()
-        status = 404
+        raise HttpError(description="Question not found.",
```
Comment (odrusso) [inline app/solutions.py:35]: Nice
I guess you could extent this to be `f"Question {this.question_type} {this.question_id} not found"`

**Review by odrusso — APPROVED** [review]
Groovy

File: app/app.py (hunk 2 above)
Comment (ashtonmoomoo) [reply app/app.py:42]: well it is part of the object, but wanted to keep in line with the rest of the functions returning `body, status`

**Pull request merged**